### PR TITLE
Move atomic harden from atomic to OpenSCAP Daemon

### PR DESCRIPTION
**Pull request by jan-cerny**
In this pull request we introduce the code that should be introduced into project Atomic to support Atomic hardening feature. We split off the code to a separate script that will be installed on host to /etc/atomic.d/scripts. The script will be shipped in "openscap" image.

### CODE REVIEW

**Comment by jan-cerny** [comment]
Jenkins has a big exception.

File: container/remediate.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    results_dir = None
+    for arg in sys.argv[1:]:
+        key, _, value = arg.partition("=")
+        if key == "id":
+            image_id = value
+        if key == "results_dir":
+            results_dir = value
+    if not image_id or not results_dir:
+        sys.stderr.write("Missing or wrong arguments.\n")
+        sys.stderr.write("Usage: ./remediate.py id=IMAGE_ID results_dir=RESULTS_DIR\n")
```
Comment (mpreisler) [inline container/remediate.py]: Could you please use argparse?

File: container/remediate.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        )
+    finally:
+        shutil.rmtree(temp_dir)
+
+
+if __name__ == "__main__":
+    image_id = None
+    results_dir = None
+    for arg in sys.argv[1:]:
+        key, _, value = arg.partition("=")
```
Comment (mpreisler) [inline container/remediate.py]: this whole logic would disappear if it used argparse

File: container/remediate.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        dockerfile_path = os.path.join(temp_dir, "Dockerfile")
+        with open(dockerfile_path, "w") as f:
+            f.write("FROM " + target_id + "\n")
+            f.write("COPY fix.sh /\n")
+            f.write("RUN chmod +x /fix.sh; /fix.sh\n")
+
+        try:
+            build_output_generator = client.build(
+                path=temp_dir,
+                nocache=True
```
Comment (mpreisler) [inline container/remediate.py]: I know we discussed this before, but could you please comment why you are using nocache=True here? It is not obvious.

File: container/remediate.py (hunk 1 above)
Comment (jan-cerny) [reply container/remediate.py]: I can't because I have to use key=value pairs instead of command line options.

File: container/remediate.py (hunk 2 above)
Comment (jan-cerny) [reply container/remediate.py]: Yes, it would.

File: container/remediate.py (hunk 3 above)
Comment (jan-cerny) [reply container/remediate.py]: OK, thanks for suggestion, I'll provide a comment.

File: container/remediate.py (hunk 1 above)
Comment (jan-cerny) [reply container/remediate.py]: Aha, in meanwhile they changed it in Atomic, I can use argparse now as I did originally, so I didn't have to rework it. I will revert the patch with reworking and go back to original.

File: container/remediate.py (hunk 2 above)
Comment (jan-cerny) [reply container/remediate.py]: :-) :-)

**Comment by jan-cerny** [comment]
I have removed 3b5ec76 from this PR to go back to original version.

**Comment by yuumasato** [comment]
The code looks good.
But can the mixed use of terms like harden and remediate cause confusions?
The changes on Atomic side don't mention harden, just remediate.

**Comment by mpreisler** [comment]
ACK

**Pull request merged**